c10/xpu/XPUStream.cpp

```diff
@@ -245,13 +251,49 @@ XPUStream getStreamFromPool(const bool isHighPriority, DeviceIndex device) {
   return getStreamFromPool(priority, device);
 }
 
+/*
+ * Note [External XPU Stream]
+ *
+ * An external XPUStream is a wrapper around an external SYCL queue that was not
+ * created by PyTorch. This design enables interoperability with other libraries
+ * by allowing PyTorch to work seamlessly with SYCL queues created outside of
+ * its control.
+ *
+ * Key design requirements include:
+ *   1. Allowing retrieval of the its SYCL queue from the external XPUStream.
+ *   2. Supporting conversion between an external XPUStream and a `c10::Stream`.
+ *   3. Ensuring compatibility with the `get/setCurrentXPUStream` methods.
+ *   4. Enabling memory caching allocation through the external XPUStream.
+ *
+ * To address requirements (1) and (2), we associate the external SYCL queue
+ * pointer with the `stream_id`. It is the user's responsibility to ensure that
+ * the referenced SYCL queue remains alive while the corresponding XPUStream, or
+ * any c10::Stream derived from it, is in use.
+ *
+ * However, this approach introduces the following limitations:
+ *
+ *   1. Different SYCL queue pointers will result in distinct XPUStream objects,
+ * even if they reference the same underlying SYCL queue object.
```

Comment:
Let's make it more clear. Ponters to queue and its copies will be distinct.

Reply:
I refine this comment. I would like to empathize that the different pointers will lead to different XPUStream instances no matter if the sycl queues they dereference are equivalent.